anxcloud/resource_kubernetes_cluster.go

```diff
@@ -54,6 +54,13 @@ func resourceKubernetesClusterCreate(ctx context.Context, d *schema.ResourceData
 		EnableAutoscaling: pointer.Bool(d.Get("enable_autoscaling").(bool)),
 	}
 
+	allowlistAsInterfaces := d.Get("apiserver_allowlist").([]interface{})
+	allowlist := make([]string, 0, len(allowlistAsInterfaces))
+	for _, v := range allowlistAsInterfaces {
+		allowlist = append(allowlist, v.(string))
+	}
```

Comment:
I haven't checked whether Terraform automatically validates that there are only strings in the `allowListAsInterfaces` or if I as an end user could sneak in some other data types in here as well.

If so, we should gracefully handle the error case instead of panicking (which would happen if `v` is not a string). If there's some handling by Terraform, please write that down above the newly added code, so that future readers are aware of this.

Reply:
As per discussion on MM: Terraform runs a validation ahead of it, no further checking needed.

Reply:
As discussed: Terraform is checking for validity before executing (i.e. `apiserver_allowlist` must be a list of strings), we don't need explicit error handling here.